### ISSUE TITLE
Huginn for Windows: image operating system "linux" cannot be used on this platform.

### ISSUE THREAD

**Issue by windyy01888** [huginn/huginn#2284]
I installed Docker for windows, input the command of "docker run -it -p 3000:3000 huginn/huginn" and switched Docker to windows container, but it occurred that "image operating system "linux" cannot be used on this platform."

What should I do?

**Comment by dsander** [comment]
I think you need to switch you docker installation to linux containers: https://docs.docker.com/docker-for-windows/#switch-between-windows-and-linux-containers

**Comment by windyy01888** [comment]
> I think you need to switch you docker installation to linux containers: https://docs.docker.com/docker-for-windows/#switch-between-windows-and-linux-containers

Thank you! The code could run this time, but it doesn't stop. I mean when I input the name of another image, it will download, pull the file, and show a message like "This message shows that your installation appears to be working correctly" after some time. But the image of huginn seems to continue forever after downloading and pulling, without showing the message. In addition, I cannot open the address "http://docker-machine ip:3000" in the browser.

Is there any solution for this situation?

**Comment by dsander** [comment]
Can you access the output of the Huginn container you started? You might only be able to access it via `http://127.0.0.1:3000`

**Comment by windyy01888** [comment]
No. The agents don't work.
![default](https://user-images.githubusercontent.com/39111058/39859566-d260d81c-546c-11e8-9789-9788029f94c4.PNG)
![1](https://user-images.githubusercontent.com/39111058/39859567-d294fcaa-546c-11e8-8ba3-4b0d612b8555.PNG)



.

**Comment by dsander** [comment]
Are you referring to the `Working?` column? That will change once your Agents create and receive Events. Try running one of the `WebsiteAgent`s manually, if they emit Events the working status will change.

**Comment by windyy01888** [comment]
> Are you referring to the Working? column? That will change once your Agents create and receive Events. Try running one of the WebsiteAgents manually, if they emit Events the working status will change.

Yes. I tried to run one of the agents and here is the result. I think there's something wrong.

The code is from the sharing of others, which has been tried successfully by many people. So I wonder if the problem comes from the installation process of the huginn image. Is it a normal condition that the installation process goes constantly without stop?

![default](https://user-images.githubusercontent.com/39111058/39918959-bb2c8a6e-5544-11e8-8fe1-c486ffa37e11.PNG)

**Comment by dsander** [comment]
> Is it a normal condition that the installation process goes constantly without stop?

That does not sound normal, can you post the output of the docker image here or to a pastebin?

**Comment by windyy01888** [comment]
>That does not sound normal, can you post the output of the docker image here or to a pastebin?

Here is the output after 30 minutes running. https://pastebin.com/cR8jAHFb

**Comment by dsander** [comment]
That output looks pretty normal to me. Did you try to manually run the WebsiteAgents from the default scenario to check if they are generating Events?
For me the default setup worked well with Kitematic under Windows 10.

**Comment by windyy01888** [comment]
> That output looks pretty normal to me. Did you try to manually run the WebsiteAgents from the default scenario to check if they are generating Events?
For me the default setup worked well with Kitematic under Windows 10.

So the problem may come from my code ha? I will try as you suggested later.
Now a new question arises that I cannot find my agent. The address 'http://127.0.0.1:3000' you gave me before cannot be opened now, neither can the address given in the official introduction.
![default](https://user-images.githubusercontent.com/39111058/40768963-5913872e-64e9-11e8-97f2-4541d62a523f.PNG)

**Comment by windyy01888** [comment]
> That output looks pretty normal to me. Did you try to manually run the WebsiteAgents from the default scenario to check if they are generating Events?
For me the default setup worked well with Kitematic under Windows 10.

First say thank you for your help, my agent can work finally :-) And now there's a new question that the url captured from the website cannot be read, I don't know what it means by 'non-HTTP url'.

Fig.1 is the url I captured and fig.2 is the output.
![1](https://user-images.githubusercontent.com/39111058/40953853-30a040b0-68b4-11e8-9057-84c8f2e3660d.PNG)
![2](https://user-images.githubusercontent.com/39111058/40953854-30dd3768-68b4-11e8-8296-5ff0d524e2e8.PNG)

**Comment by dsander** [comment]
How did you configure the Agent? If you want to use `url_from_event` you need to use an Event when dry running the Agent. The error looks like you put `url_from_event` into the `url` key. Try setting `url_from_event` to `{{url}}` and remove the `url` configuration option.
